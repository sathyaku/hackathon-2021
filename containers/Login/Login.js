import React, { useState, useEffect } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import Router from 'next/router';
import Button from '@/ui/Button';
import ContainerWrap from '@/ui/Container';
import layOutStyle from './Login.style';
import { WithLabels } from 'contexts/LabelContext';
import { WithDevice } from 'contexts/DeviceContext';
import { useRouter } from 'next/router';
import FormInput from '@/ui/FormInput';

const signin = async (email, password) => {
  // axios.post(`/api/test?email=${email}&password=${password}`)
  const body = {
    email,
    password
  };
  axios({
    method: 'post',
    url: '/api/test',
    data: body
  }).then(
    (res) => {
      Router.push('/dashboard');
    },
    () => {}
  );
};

const LoginContainer = ({ labels }) => {
  const [userData, setUserData] = useState({
    email: '',
    password: '',
    error: ''
  });
  const router = useRouter();

  const { LOGIN_PAGE_HEADER } = labels;

  async function handleSubmit() {
    setUserData({ ...userData, error: '' });

    const { email } = userData;
    const { password } = userData;

    try {
      await signin(email, password);
    } catch (error) {
      //   console.error(error);
      setUserData({ ...userData, error: error.message });
    }
  }

  useEffect(() => {
    router.prefetch('/dashboard');
  }, []);

  return (
    <ContainerWrap direction="column">
      <div className="login">
        <h1> {LOGIN_PAGE_HEADER} </h1>
        <form>
          <FormInput
            label="Email"
            onChangeHandler={(event) =>
              setUserData({ ...userData, email: event.target.value })
            }
            value={userData.email}
            name="email"
            id="email"
            type="text"
          />

          <FormInput
            label="Password"
            onChangeHandler={(event) =>
              setUserData({ ...userData, password: event.target.value })
            }
            value={userData.email}
            name="password"
            id="password"
            type="password"
          />

          <div className="btn-container">
            <Button
              key="login-btn"
              id="loginButton"
              label="Login"
              strech={'full'}
              size="medium"
              onClick={handleSubmit}
            />
          </div>
        </form>
      </div>
      <style jsx>{layOutStyle}</style>
    </ContainerWrap>
  );
};

LoginContainer.defaultProps = {
  labels: {}
};

LoginContainer.propTypes = {
  labels: PropTypes.object
};

export default WithLabels(LoginContainer);
export { LoginContainer };
