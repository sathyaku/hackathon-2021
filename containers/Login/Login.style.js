import css from 'styled-jsx/css';

const layOutStyle = css`
  .login {
    @mixin flexFromStart {
      position: fixed;
      margin: 30vh auto;
      width: 440px;
      padding: 20px 40px;
      border: 1px solid #ccc;
      border-radius: 8px;
      box-shadow: 0 2px 7px #dfdfdf;
      flex-direction: column;

      h1 {
        font-size: 18px;
        font-weight: 600;
        margin: 30px 0;
      }
      @mixin smallMobileOnly {
        width: 100%;
        height: 100vh;
        margin: 0 auto;
      }
    }
  }
  form {
    @mixin flexFromStart {
      flex-flow: column;
      gap: 1ch;
      width: 100%;

      .btn-container {
        width: 100%;
        margin: 20px auto;
      }
    }
  }

  label {
    font-weight: 600;
  }
  .error {
    margin: 0.5rem 0 0;
    color: brown;
  }
`;

export default layOutStyle;
