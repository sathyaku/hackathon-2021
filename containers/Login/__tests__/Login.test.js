import React from 'react';
import { mount } from 'enzyme';

import { LoginContainer } from '../Login';

describe('Login Container', () => {
  it('should render Login', () => {
    const wrapper = mount(
      <LoginContainer labels={{ LOGIN_PAGE_HEADER: 'User Login' }} />
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
