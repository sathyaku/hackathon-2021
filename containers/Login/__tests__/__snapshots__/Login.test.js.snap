// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container should render Login 1`] = `
"<LoginContainer labels={{...}}>
  <div className=\\"jsx-3109774058 container\\">
    <h1 className=\\"jsx-3109774058\\">
       
      User Login
       
    </h1>
    <div className=\\"jsx-3109774058 login\\">
      <form className=\\"jsx-3109774058\\">
        <label htmlFor=\\"email\\" className=\\"jsx-3109774058\\">
          Email
        </label>
        <input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" onChange={[Function: onChange]} className=\\"jsx-3109774058\\" />
        <label htmlFor=\\"password\\" className=\\"jsx-3109774058\\">
          Password
        </label>
        <input type=\\"password\\" id=\\"password\\" name=\\"password\\" value=\\"\\" onChange={[Function: onChange]} className=\\"jsx-3109774058\\" />
        <button id=\\"loginButton\\" type=\\"button\\" onClick={[Function: handleSubmit]} className=\\"jsx-3109774058 btn\\">
          Login
        </button>
      </form>
    </div>
    <JSXStyle id=\\"3109774058\\" />
  </div>
</LoginContainer>"
`;
